pkg/operator/targetconfigcontroller/targetconfigcontroller.go

```diff
@@ -301,7 +297,7 @@ func managePod_v311_00_to_latest(ctx context.Context, configMapsGetter corev1cli
 	configMap.Data["version"] = version.Get().String()
 	appliedConfigMap, changed, err := resourceapply.ApplyConfigMap(ctx, configMapsGetter, recorder, configMap)
 	if changed && len(config.Spec.Policy.Name) > 0 {
```

Comment:
With `manageKubeSchedulerConfigMap_v311_00_to_latest` returning error when `len(config.Spec.Policy.Name) > 0`, the kube-scheduler container will keep crash looping since the configmap with the profile will be missing (assuming the `scheduler/cluster` object was specified with `.spec.policy.name` set during the cluster provisioning). I wonder if you took this case into account and whether it would make more sense to move `len(config.Spec.Policy.Name) > 0` check alongside the `config, err := configSchedulerLister.Get("cluster")` line at the beginning of the `managePod_v311_00_to_latest` function and return an error as well so the pod does not get created and kept crash looping until the policy field is cleared?

The case covers the bootstrapping phase in which (when `.spec.policy.name` is not empty) the installation fails. So it is unlikely any admin will update the `scheduler/cluster` object rather than running the installation again. So, the net benefit for the normal installation is quite low. On the other hand when the hypershift topology is used (with one cluster hosting many control planes), postponing the creation of the kube-scheduler pod might safe the step of debugging why the pod is crash looping (depending on how the control plan is provisioned).

Reply:
My comment is more for the debugging purposes than the functional/conceptual ones. The operator will go degraded when the policy field is set.